### DISCUSSION TITLE
ValidateFile - regex for allowTypes , modifiers no longer supported ?

### DISCUSSION THREAD

**Discussion by cyanotyp** [primefaces/community#2619] [PrimeFaces] [answered]
Hi all,
I have a question if in PF14 the modifiers for **RegEx** on e.g. `<p:validatefile allowTypes` are no longer supported or if this is a bug .
- My simple useCase was in PF13  `allowTypes="/(\.|\/)(xlsx)$/i"` 
- the **/i** modifier was added since some users preferred to use file extension in capitals e.g myfile.XLSX
- This worked fine in PF13, but in 14 a file with caps extension now shows the "unsupported file type" message.

I took a little time to check the PF source code and found in **/resources/primefaces/validation/validation.validators.js**
`        // but as we parse the data-attribute from string to RegEx object, we must remove leading and ending slashes
                    var transformedAllowtypes = allowtypes.substring(1, allowtypes.length - 1);`

So a RegEx always is always stripped by 1 char at both ends, so my  **/i** modifier is stripped.

My basic question is: Is this intended or a PrimeFaces bug? Based on the answer I have to either change my regex or file a bug :)

**Comment by melloware** [comment]
This looks like a regression bug to me if you want to report it please on the Issues page

**Reply by cyanotyp** [reply, marked as answer]
filed PF bug https://github.com/primefaces/primefaces/issues/12551 :)
